### PR TITLE
IJSDK-600 - Add SDK code and doc style guides

### PR DESCRIPTION
**Pull request by JohnHake**
I'll merge after review.

### CODE REVIEW

File: intro/sdk_code_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+A consistent structure for an SDK code sample's `build.gradle` file is important for clarity and is based on the default produced by the project wizard. 
+Comments in SDK code sample `build.gradle` files should only draw attention to the parts of the Gradle configuration that are unique for a plugin.
+
+For SDK code samples a few alterations are needed to the default build.gradle file produced by the plugin wizard:
+* Add the following statements to the [Setup DSL](https://github.com/JetBrains/gradle-intellij-plugin/blob/master/README.md#setup-dsl) (`intellij{}`) section:
+  ```groovy
+      // Prevents patching <idea-version> attributes in plugin.xml
+      updateSinceUntilBuild = false 
+      // Define a shared sandbox directory for running code sample plugins within an IDE.
+      sandboxDirectory = file("${project.projectDir}/../_Build/idea-sandbox")
```
Comment (YannCebron) [inline intro/sdk_code_guidelines.md]: where does `_Build` come from?

File: intro/sdk_code_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+* `<change-notes>` is an ordered list by version numbers with a brief description of changes for each version.
+* `<vendor>` Set the value to `IntelliJ Platform SDK`.
+  Set the attributes:
+  * `email` to a placeholder value `"faux-email@jetbrains.com"` 
+  * `url` to the JetBrains plugin repository `"https://plugins.jetbrains.com"`
+* Except for [special cases](/basics/plugin_structure/plugin_configuration_file.md) these elements are not used in SDK plugins.
+  * `<depends>`  
+  * `<helpset>` 
+  * `<resource-bundle>`
+* `<application-components>` Deprecated and should not be used for new plugins.
```
Comment (YannCebron) [inline intro/sdk_code_guidelines.md]: `<application-components>` is not deprecated, but indeed use is discouraged (whenever possible)

File: intro/sdk_code_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+* `<module-components>` Omit if not specifically used.
+* `<project-components>` Omit if not specifically used.
+* `<extensionPoints>` Generally not used by SDK plugins.
+* `<actions>` Omit if not specifically used.
+* `<extensions>` Omit if not specifically used.
+
+## Testing
+IntelliJ Platform SDK code samples should be built and tested against the `since-build` version.
+
+Code samples should build cleanly, with no warnings or errors, and new code samples should pass all IntelliJ IDEA code inspections.
```
Comment (YannCebron) [inline intro/sdk_code_guidelines.md]: "inspections" as in "all by default enabled inspections" ?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -88,202 +74,54 @@ This is made easier if you use [Chocolatey](https://chocolatey.org), a package m
...
+If changes are made to the submodule, they should be made on a branch to a clone, and a Pull Request sent. 
+Changes can be made and committed, and the hosting repository will need to commit a pointer to the current version of the submodule.
 
 If there are any problems with the `sdkdocs-template`, please [raise an issue](https://github.com/JetBrains/sdkdocs-template/issues).
 
+## Creating IntelliJ Platform SDK Content
+Content contributions to the IntelliJ Platform SDK are welcome.
+Please download or clone the open source SDK project from [GitHub](https://github.com/JetBrains/intellij-sdk-docs), make additions or changes, and submit a pull request.
+Before creating or altering content, please consult these guides:
+* [SDK Code Sample Guidelines](intro/sdk_code_guidelines.md).
```
Comment (YannCebron) [inline CONTRIBUTING.md]: order of Code vs Docs links is swapped with menu

File: intro/sdk_code_guidelines.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+There is only one _basic_ sample per IntelliJ Platform API area.
+
+For _advanced_ code samples, the name should reflect the complex functionality delivered by the plugin rather than the IntelliJ Platform APIs.
+Advanced samples will be cross-referenced to the IntelliJ Platform APIs demonstrated in the sample.
+
+Regardless of basic or advanced, an SDK plugin name is also known as the `Artifact ID` in the Gradle plugin wizard.
+
+## Plugin Copyright Statements
+Use the standard intellij-community copyright notice in all sample plugins authored by JetBrains:  
+   "Copyright 2000-$today.year JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file."
```
Comment (YannCebron) [inline intro/sdk_code_guidelines.md]: format as code block for easy c/p

File: intro/sdk_code_guidelines.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+  * `since-build` attribute to the earliest compatible build number of the IntelliJ Platform.
+  * `until-build` attribute to an empty string `""`.
+    SDK code samples are reviewed with every major release to ensure compatibility with the latest IntelliJ Platform.
+* `<description>` is a succinct explanation of what is being demonstrated and how a user would access the functionality.
+* `<change-notes>` is an ordered list by version numbers with a brief description of changes for each version.
+* `<vendor>` Set the value to `IntelliJ Platform SDK`.
+  Set the attributes:
+  * `email` to a placeholder value `"faux-email@jetbrains.com"` 
+  * `url` to the JetBrains plugin repository `"https://plugins.jetbrains.com"`
+* Except for [special cases](/basics/plugin_structure/plugin_configuration_file.md) these elements are not used in SDK plugins.
```
Comment (YannCebron) [inline intro/sdk_code_guidelines.md]: I'd propose to cut down the rest of this list by simply stating that all other elements are to be present only for specific plugins.

File: intro/sdk_code_guidelines.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+  * FIX corresponds to changes that fix problems without significant refactoring.
+* `<idea-version/>` Set the attributes:
+  * `since-build` attribute to the earliest compatible build number of the IntelliJ Platform.
+  * `until-build` attribute to an empty string `""`.
+    SDK code samples are reviewed with every major release to ensure compatibility with the latest IntelliJ Platform.
+* `<description>` is a succinct explanation of what is being demonstrated and how a user would access the functionality.
+* `<change-notes>` is an ordered list by version numbers with a brief description of changes for each version.
+* `<vendor>` Set the value to `IntelliJ Platform SDK`.
+  Set the attributes:
+  * `email` to a placeholder value `"faux-email@jetbrains.com"`
```
Comment (YannCebron) [inline intro/sdk_code_guidelines.md]: maybe get rid of "email" and just use url

File: intro/sdk_code_guidelines.md (hunk 1 above)
Comment (JohnHake) [reply intro/sdk_code_guidelines.md]: "_Build" = directory that holds the sandbox directory. Appears at the top of the code_samples directory for convenience. code_samples/action_basics/build is the default gradle output directory where everything else is located.

File: intro/sdk_code_guidelines.md (hunk 5 above)
Comment (JohnHake) [reply intro/sdk_code_guidelines.md]: agreed

File: intro/sdk_code_guidelines.md (hunk 2 above)
Comment (JohnHake) [reply intro/sdk_code_guidelines.md]: agreed.

File: intro/sdk_code_guidelines.md (hunk 3 above)
Comment (JohnHake) [reply intro/sdk_code_guidelines.md]: Yes, I will clarify.

File: CONTRIBUTING.md (hunk 4 above)
Comment (JohnHake) [reply CONTRIBUTING.md]: I'll update the CONTRIBUTING doc to match the _SUMMARY menu.

File: intro/sdk_code_guidelines.md (hunk 6 above)
Comment (JohnHake) [reply intro/sdk_code_guidelines.md]: good idea.

File: intro/sdk_code_guidelines.md (hunk 7 above)
Comment (JohnHake) [reply intro/sdk_code_guidelines.md]: Is the email redundant to information at plugins.jetbrains.com?

**Pull request merged**